### ISSUE TITLE
archive tarball build failure due to missing ChangeLog

### ISSUE THREAD

**Issue by ilovezfs** [andreabolognani/cattle#4]
Full build log: https://gist.github.com/ilovezfs/4898eac0e41c7dbab7b2f6c728c80ee8

The error is

```
/usr/bin/install -c -m 644 AUTHORS.markdown ./ChangeLog ChangeLog.pre-0.9.2 COPYING NEWS.markdown README.markdown TODO.markdown '/usr/local/Cellar/cattle/1.2.1/share/doc/cattle-1.0'
install: ./ChangeLog: No such file or directory
make[2]: *** [install-dist_docDATA] Error 71
make[1]: *** [install-am] Error 2
make: *** [install-recursive] Error 1
```

Building from the tag itself checkout from a clone of the git repository succeeds.

The cause seems to be

```
    if test -d "./.git"; \
    then \
      GIT_DIR="./.git" \
      ./missing \
        --run git log --color=never --stat \
      >ChangeLog.tmp \
      && mv -f ChangeLog.tmp ChangeLog \
      || ( rm -f -f ChangeLog.tmp; false ) \
    fi
```

doesn't run because there is no `.git` directory in the GitHub source archive tarball.

**Comment by andreabolognani** [comment]
Where did you get the release archive?

I'm guessing you got it from the project's "Releases" tab here on GitHub, which seems to match the unpack output: a plain dump of the contents of the repository, including `autogen.sh` and a bunch of `.gitignore` files, which are clearly of no use to someone building from a release archive.

Apparently GitHub believes that deleting the `.git` directory and packing the leftovers into a `.tar.gz` is enough to turn a repository clone into a proper release, and that creating such a "release" automatically for every tag is a good idea. I can't seem to find a way to disable this annoying behavior, either.

Luckily, Cattle uses a build system that supports creating proper releases that don't dump irrelevant files on the end user's system or suffer from the issue you reported :)

You can find the proper release archives on [the relevant page on Cattle's website](http://kiyuko.org/software/cattle/releases). Let me know if you have any trouble using those; in the meantime, I'll try to convince GitHub to replace their "release" archives with the proper ones.

**Comment by ilovezfs** [comment]
>Where did you get the release archive? I'm guessing you got it from the project's "Releases" tab here on GitHub ...

Yes, correct.

You can post a release tarball for a given tag that is distinct from the auto-generated source archive tarball by following these instructions: https://help.github.com/articles/creating-releases/

For examples, see
https://github.com/tmux/tmux/releases
https://github.com/commercialhaskell/stack/releases

**Comment by andreabolognani** [comment]
Hi,

I investigated turning off the feature (not possible) or removing existing "releases" (not possible unless you remove the tags as well).

I could add proper release archives and release notes to existing tags, and it would certainly look quite cool; however:

  * the broken downloads will still be present besides the proper release archive;
  * as soon as I do that for a single tag, I will feel compelled to do the same for all tags in the repository, and then for all repositories I have mirrored on GitHub, which would be a lot of boring work.

In light of this, I decided to sidestep the issue by adding some information on how to obtain Cattle to the `README` file (see 8af9f0d6c60ff4c96e8ee5766aecae8e49caf522); hopefully that will be enough to inform users going forward. If not, I will still be able to point them to this discussion ;)

Thanks for reporting the issue though! Hopefully you're not having any trouble now that you've switched to proper release archives.
